cmd/node/list.go

```diff
@@ -92,52 +92,60 @@ func runList(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn) err
 	})
 	switch listOpts.output {
 	case "json":
-		bs, err := json.MarshalIndent(nodes, "", "  ")
-		if err != nil {
-			return err
-		}
-		_, err = fmt.Fprintln(cmd.OutOrStdout(), string(bs))
-		return err
+		return jsonOutput(cmd, nodes)
 	case "table", "wide":
-		tw := tabwriter.NewWriter(cmd.OutOrStdout(), 2, 0, 3, ' ', 0)
-		fmt.Fprint(tw, "NAME\tSTATUS\tAGE\tFLOWS/S\tCURRENT/MAX-FLOWS")
-		if listOpts.output == "wide" {
-			fmt.Fprint(tw, "\tVERSION\tADDRESS\tTLS")
-		}
-		fmt.Fprintln(tw)
+		return tableOutput(cmd, nodes)
+	default:
+		return fmt.Errorf("unknown output format: %s", listOpts.output)
+	}
+}
 
-		for _, n := range nodes {
-			age := notAvailable
-			flowsPerSec := notAvailable
-			if uptime := time.Duration(n.GetUptimeNs()).Round(time.Second); uptime > 0 {
-				age = uptime.String()
-				flowsPerSec = fmt.Sprintf("%.2f", float64(n.GetSeenFlows())/uptime.Seconds())
-			}
-			flowsRatio := notAvailable
-			if maxFlows := n.GetMaxFlows(); maxFlows > 0 {
-				flowsRatio = fmt.Sprintf("%d/%d (%6.2f%%)", n.GetNumFlows(), maxFlows, (float64(n.GetNumFlows())/float64(maxFlows))*100)
-			}
-			version := notAvailable
-			if v := n.GetVersion(); v != "" {
-				version = v
-			}
-			fmt.Fprint(tw, n.GetName(), "\t", strings.Title(nodeStateToString(n.GetState())), "\t", age, "\t", flowsPerSec, "\t", flowsRatio)
-			if listOpts.output == "wide" {
-				tls := notAvailable
-				if t := n.GetTls(); t != nil {
-					tls = "Disabled"
-					if t.Enabled {
-						tls = "Enabled"
-					}
+func tableOutput(cmd *cobra.Command, nodes []*observerpb.Node) error {
```

Comment:
nit: this function could take an `io.Writer` as first parameter instead of a cobra command.

Reply:
Good point, this will reduce the setup in each test as well. Taking an `io.Writer` as an input arg is easier for future usage too. I implemented it: https://github.com/cilium/hubble/pull/496/commits/44b11fb232f1ebe977ca51e1c66ced1dda45b7bd